Allure_results_sprint_7

Mswahili88 opened the pull request:
adding allure report

Mswahili88 committed:
add allure report

Mswahili88 committed:
incorrectly unedrstood the task and pushed allure-report only. adding the rest of the tests

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,28 @@
+    response_id = requests.post(urls.URL_LOGIN, data=TestDataBody.BODY_CHECK_LOGIN)
+    id_number = response_id.json()["id"]
+    response_delete = requests.delete(f"{urls.URL_DELETE_COURIER}{id_number}")
+    return response_delete
```
Нужно исправить: этого тут быть не должно, оно работать не будет

AlexKitIt commented on tests/tests.py:
```diff
@@ -0,0 +1,63 @@
+        response = requests.post(urls.URL_CREATE_COURIER, data=TestDataBody.BODY_WITHOUT_LOGIN)
+        assert response.status_code == 400 and response.json()["message"] == 'Недостаточно данных для создания учетной записи'
+
+class TestLoginCourier:
```
Нужно исправить: каждый класс должен быть в отдельном модуле

AlexKitIt commented on urls.py:
```diff
@@ -0,0 +1,5 @@
+URL_CREATE_COURIER = 'https://qa-scooter.praktikum-services.ru/api/v1/courier'
+URL_LOGIN = 'https://qa-scooter.praktikum-services.ru/api/v1/courier/login'
+URL_DELETE_COURIER = 'https://qa-scooter.praktikum-services.ru/api/v1/courier/'
+URL_CREATE_ORDER = 'https://qa-scooter.praktikum-services.ru/api/v1/orders'
+URL_GET_ALL_ORDERS = 'https://qa-scooter.praktikum-services.ru/api/v1/orders'
```
Можно лучше: урлы стоит конкатенировать с базовым, так будет проще поддерживать несколько стендов

Mswahili88 committed:
remade the conftest.py, added helper.py and remade some tests. URLS is corrected. Division of the tests by different modules is done

AlexKitIt commented on helper.py:
```diff
@@ -0,0 +1,45 @@
+    @allure.step('Создание рандомных логина, пароля и имени')
+    def create_random_login_password():
+
+        def generate_random_string(length):
+            letters = string.ascii_lowercase
+            random_string = ''.join(random.choice(letters) for i in range(length))
+            return random_string
```
Нужно исправить: метод копипастится второй раз, так делать нельзя, если он переиспользуется, то он не должен быть вложенным

AlexKitIt commented on helper.py:
```diff
@@ -0,0 +1,45 @@
+class TestMethodsHelper:
+    @staticmethod
+    @allure.step('Регистрация нового курьера с рандомными данными и возврат этих данных')
+    def register_new_courier_and_return_login_password():
+        def generate_random_string(length):
+            letters = string.ascii_lowercase
+            random_string = ''.join(random.choice(letters) for i in range(length))
+            return random_string
+
+        login_pass = {}
+
+        login = generate_random_string(10)
+        password = generate_random_string(10)
+        first_name = generate_random_string(10)
+
+        payload = {"login": login, "password": password, "firstName": first_name}
```
Нужно исправить: большая часть этого метода теперь описана в фикстуре create_random_login_password, копипастить не надо, надо вызывать ее и дополнять отправкой запроса

AlexKitIt commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,27 @@
+    @allure.description('Проверяем успешное создание курьера и в ответ статус 201 и тело ответа {"ok":true}')
+    def test_create_courier(self, default_courier):
+        payload = default_courier
+        id_number = (requests.post(urls.URL_BASE + urls.URL_LOGIN, data=payload)).json()["id"]
+        requests.delete(f"{urls.URL_BASE}{urls.URL_DELETE_COURIER}{id_number}")
```
Нужно исправить: что за логика тут описана? Зачем нужно создвать курьера, удалять его, а потом снова создавать??

Mswahili88 replied:
@AlexKitIt День добрый! В фикстуре используется метод создания курьера и возврата его данных. То есть, курьер уже создан. Если я беру оттуда данные в эту проверку, то мне же сначала нужно удалить этого курьера и сразу же проверить создание нового с теми же данными, так как они более "не заняты". Или логика не верна?

AlexKitIt commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,27 @@
+    def test_courier_same_name(self, default_courier):
+        payload = default_courier
+        response = requests.post(urls.URL_BASE + urls.URL_CREATE_COURIER, data=payload)
+        assert response.status_code == 409 and response.json()["message"] == 'Этот логин уже используется. Попробуйте другой.'
```
Можно лучше: тексты ответов стоит хранить снаружи в data модулях для поддержания и актуализации

Mswahili88 committed:
1. Corrected helper-methods and choose only one variant
2. Took away texts of mistakes into the data.py
3. Corrected the first test -"Creating the courier"
